internal/runtime/client/client.go

```diff
@@ -153,6 +163,98 @@ func (c *client) Unregister(extensionConfig *runtimev1.ExtensionConfig) error {
 	return nil
 }
 
+// CallAllExtensions calls all the ExtensionHandlers registered for the hook.
+// The ExtensionHandler are called sequentially. The function exits immediately after any of the ExtensionHandlers return an error.
+// See CallExtension for more details on when an ExtensionHandler returns an error.
+// The aggregate result of the ExtensionHandlers is updated into the response object passed to the function.
+func (c *client) CallAllExtensions(ctx context.Context, hook catalog.Hook, request runtime.Object, response runtimehooksv1.Response) error {
+	gvh, err := c.catalog.GroupVersionHook(hook)
+	if err != nil {
+		return errors.Wrap(err, "failed to compute GroupVersionHook")
+	}
+	registrations, err := c.registry.List(gvh.GroupHook())
+	if err != nil {
+		return errors.Wrapf(err, "failed to retrieve ExtensionHandlers for %s", gvh.GroupHook())
+	}
+	responses := []runtimehooksv1.Response{}
+	for _, registration := range registrations {
+		// Creates a new instance of the response parameter.
+		tmpResponse := reflect.New(reflect.TypeOf(response).Elem()).Interface().(runtimehooksv1.Response)
+		err = c.CallExtension(ctx, hook, registration.Name, request, tmpResponse)
+		// If one of the extension handlers fails lets short-circuit here and return early.
+		if err != nil {
+			return errors.Wrapf(err, "ExtensionHandler %s failed", registration.Name)
+		}
+		responses = append(responses, response)
+	}
+	if err := aggregateResponses(responses); err != nil {
+		return errors.Wrap(err, "failed to aggregate responses")
+	}
+	return nil
+}
+
+func aggregateResponses(responses []runtimehooksv1.Response) error {
+	// TODO:(killianmuldoon) implement proper aggregation logic.
+	return nil
+}
+
+// CallExtension make the call to the extension with the given name.
+// The response object passed will be updated with the response of the call.
+// An error is returned if the extension is not compatible with the hook.
+// If the ExtensionHandler returns a response with `Status` set to `Failure` the function returns an error
+// and the response object is updated with the response received from the extension handler.
+//
+// FailurePolicy of the ExtensionHandler is used to handle errors that occur when performing the external call to the extension.
+// - If FailurePolicy is set to Ignore, the error is ignored and the response object is updated to be the default success response.
+// - If FailurePolicy is set to Fail, an error is returned and the response object may or may not be updated.
+// Nb. FailurePolicy does not affect the following kinds of errors:
+// - Internal errors. Examples: hooks is incompatible with ExtensionHandler, ExtensionHandler information is missing.
+// - Error when ExtensionHandler returns a response with `Status` set to `Failure`.
```

Comment:
I think we should surface this in the also in the API where FailurePolicy is defined.
We should also add an explanation about how we are interpreting failure policy, which is consistent with Admission webhooks; e.g.
Failure policy helps with extensions not working consistently; however, the following type of errors are always considered blocking Failures
- misconfigurations (incompatible types) 
- when the extension explicitly report a Status Failure